app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments/discounts/DiscountsDestination.kt

```diff
@@ -146,31 +151,168 @@ internal fun DiscountsScreen(
+        title = stringResource(R.string.referrals_info_sheet_headline),
+        body = stringResource(R.string.referrals_info_sheet_body, incentive),
+      )
     }
+    HorizontalItemsWithMaximumSpaceTaken(
+      startSlot = {
+        Text(
+          stringResource(R.string.PAYMENTS_REFERRALS_INFO_TITLE),
+          modifier = Modifier.wrapContentSize(Alignment.CenterStart),
+        )
+      },
+      endSlot = {
+        Icon(
+          imageVector = Icons.Hedvig.InfoFilled,
+          tint = MaterialTheme.colorScheme.onSurfaceVariant,
+          contentDescription = null,
+          modifier = Modifier
+            .wrapContentSize(Alignment.CenterEnd)
+            .size(16.dp)
+            .clip(MaterialTheme.shapes.squircleLarge)
+            .clickable { showForeverInfoBottomSheet = true }
+            .minimumInteractiveComponentSize(),
+        )
+      },
+    )
+    Spacer(modifier = Modifier.height(16.dp))
+    HorizontalItemsWithMaximumSpaceTaken(
+      startSlot = {
+        HedvigCard(
+          shape = MaterialTheme.shapes.squircleSmall,
+          colors = CardDefaults.cardColors(
+            containerColor = MaterialTheme.colorScheme.secondaryContainedButtonContainer,
+            contentColor = MaterialTheme.colorScheme.onSecondaryContainedButtonContainer,
+          ),
+          modifier = Modifier.wrapContentSize(Alignment.TopStart),
+        ) {
+          Text(
+            text = foreverInformation.foreverCode,
+            style = MaterialTheme.typography.bodyMedium,
+            modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
+          )
+        }
+      },
+      endSlot = {
+        Text(
+          stringResource(
+            R.string.OFFER_COST_AND_PREMIUM_PERIOD_ABBREVIATION,
+            "-${foreverInformation.currentMonthlyDiscountFromForever}",
+          ),
+          textAlign = TextAlign.End,
+        )
+      },
+      spaceBetween = 8.dp,
+    )
+    Spacer(modifier = Modifier.height(16.dp))
+    VectorInfoCard(
+      text = stringResource(R.string.PAYMENTS_REFERRALS_INFO_DESCRIPTION),
+      iconColor = MaterialTheme.colorScheme.typeElement,
+      icon = Icons.Hedvig.Campaign,
+      colors = CardDefaults.outlinedCardColors(
+        containerColor = MaterialTheme.colorScheme.typeContainer,
+        contentColor = MaterialTheme.colorScheme.onTypeContainer,
+      ),
+      underTextContent = {
+        HedvigContainedSmallButton(
+          modifier = Modifier.fillMaxWidth(),
+          colors = ButtonDefaults.buttonColors(
+            containerColor = MaterialTheme.colorScheme.containedButtonContainer,
+            contentColor = MaterialTheme.colorScheme.onContainedButtonContainer,
+          ),
+          textStyle = MaterialTheme.typography.bodyMedium,
+          text = stringResource(R.string.important_message_read_more),
+          onClick = { showForeverInfoBottomSheet = true },
+        )
+      },
+    )
```

Comment:
I think on the last design critique William said that the whole list of the referrals that the member already has should also be displayed here. But I think we're not doing it before they tell if it's okay from the legal perspective to show referrals' names here?

Reply:
Yeah the names I am exactly postponing until we make a decision.
For the forever well since it turns out we're keeping it as a 5th tab I will try to see how the design evolves after that decision. I think I'll merge for now and do anything else on top of this